Add getting started doc

ebeahan opened the pull request:
### Overview

This PR introduces a `Getting Started` overview in the ECS documentation. It's intended to give a brief overview of how to adopt and use ECS aimed at newcomers to the schema. The concepts and approach borrow heavily from the original [ECS intro blog](https://www.elastic.co/blog/introducing-the-elastic-common-schema). The hope is introducing helps ECS newcomers answer the question: "where do I start with ECS?" and helps to address previous user feedback (#516).

### Structure

The doc is structured in three sections: parsing, search, and visualization. Again this borrows from the structure of the original blog post. I wanted to retain the value demonstrated in the original blog and bring it into the ECS docs; the value ECS brings to each of these three areas and how ECS improves the user experience across the stack.

#### Parsing

The first section walks through a simple, common, and well-understood event, an Apache web server log, and provides a detailed field mapping. The intention is to provide newcomers with an event mapping reference example and also priming them to understand ECS adoption into their stack may require some examination of their current data pipeline and data normalization strategy.

#### Searching

Basic example of how normalizing fields will provide a simplified search experience across diverse data sources.


#### Visualization

Again a basic example to demonstrate the value of normalized fields. Added a screenshot and `images` subdirectory to the `docs` folder. There may be additional changes required for supporting images in the documentations (this is a first for the ECS docs 😄 ), but local testing looks ok. 

### Placement

`Getting Started` is added underneath the `Using` section of the existing docs immediately following the ECS `Overview` page. This seemed the most useful for a new user (view some example real-world type application before the guidelines and conventions), but I'm open for further discussion on placement.

ebeahan commented:
@benskelker would appreciate your feedback here when you have some free cycles.

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+
+Getting started adopting ECS requires two steps. First, normalizing your data sources to align
+with the ECS field sets, and then familiarizing yourself with those fields for querying
+and aggregating your data. Let's take a look at how an event's contents maps to the ECS field
```
```suggestion
and aggregating your data. Let's take a look at how an event's content maps to the ECS field
```
Maybe: `how an event maps to ECS field sets`

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+Getting started adopting ECS requires two steps. First, normalizing your data sources to align
+with the ECS field sets, and then familiarizing yourself with those fields for querying
+and aggregating your data. Let's take a look at how an event's contents maps to the ECS field
+sets and how using a common schema for your data can simplify the search experience.
```
```suggestion
sets, and how using a common schema for your data can simplify the search experience.
```

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+[float]
+==== Translating Data Sources
+
+In order for events to align to ECS, some sort of parsing will usually be necessary
```
```suggestion
To align events to ECS, some sort of parsing will usually be necessary
```

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+
+|=====
+
+Some items to note on the mapping example:
```
```suggestion
Some items to note about this mapping example:
```

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+* As many ECS fields as possible are mapped from the original event
+* Several `event` fields are populated based on ECS categorization
+* The raw log is preserved in the `event.original` field
+* For the simplicity of this example some field sets such as `agent.*`
```
```suggestion
* For the simplicity of this example some field sets, such as `agent.*`
```

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+* Several `event` fields are populated based on ECS categorization
+* The raw log is preserved in the `event.original` field
+* For the simplicity of this example some field sets such as `agent.*`
+  and `host.*` have been omitted.
```
```suggestion
  and `host.*`, have been omitted.
```

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+source.ip:10.42.42.42
+-----------
+
+Not only does this simplify writing queries, but any saved or shared queries between
```
```suggestion
Not only does this simplify writing queries, but saved queries shared queries between
```

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+-----------
+
+Not only does this simplify writing queries, but any saved or shared queries between
+users become much more obvious. Any unfamiliar fields that align with ECS can also
+be reviewed in the <<ecs-field-reference>> to gain better understanding.
```
```suggestion
users are much more obvious. To gain familiarity with ECS fields, you can also take a look at the <<ecs-field-reference>> section.
```

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+==== Unified Visualizations
+
+With normalized data from different data sources, building insightful visualizations
+across sources is also greatly simplified. From a single, centralized dashboard, events
```
```suggestion
across sources is simple. From a single, centralized dashboard, events
```

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+
+With normalized data from different data sources, building insightful visualizations
+across sources is also greatly simplified. From a single, centralized dashboard, events
+from web servers, IDS/IPS devices, and firewalls can be aggregated, visualized, provides
+drill-down, and then pivoted from into deeper investigations. Centralized monitoring of
+diverse data sources has been greatly simplified normalizing to ECS.
```
```suggestion
from web servers, IDS/IPS devices, and firewalls can be aggregated and visualized, and enhanced with
drill-downs and pivots for delving into deeper investigations. Centralized monitoring of
diverse sources is straightforward with normalized ECS data.
```
Not sure this edit keeps your original intent.

ebeahan replied:
It does - thanks for the suggestions!

webmat commented:
Thanks @benskelker for the detailed feedback as well :-)

benskelker commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+
+Adopting ECS is a two-step process. First, normalizing your data sources to align
+with the ECS field sets, and then familiarizing yourself with those fields for querying
+and aggregating your data. Let's take a look at how an event's contents can be mapped to
```
@webmat - "contents" is correct (like contents of a folder)

webmat replied:
TIL about [content vs contents](https://getproofed.com/writing-tips/word-choice-content-vs-contents/) ☝🏼

Thanks for confirming, Ben

jamiehynds commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+[[ecs-getting-started]]
+=== Getting Started
```
++

Maybe we could start with a brief explanation of what ECS and the benefits of mapping your events to ECS? e.g. ECS defines a [common set of fields](https://www.elastic.co/guide/en/ecs/current/ecs-field-reference.html) to be used when storing data in Elasticsearch. The goal of ECS is to enable and encourage users of Elastic to normalize event data in order to better analyze, visualize and correlate the data represented in their events. 

Agree with Mat that we should make users aware of our out-of-the-box ECS compliant integrations - should link to this page: https://www.elastic.co/integrations

DanRoscigno commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,191 @@
+
+
+[float]
+==== Simplified Search
```
I agree, a short intro to the reasons helps frame the rest of the doc.  Here is how I start an email to encourage use of ECS by new customers:

> The Elastic Common Schema (ECS) is an open source specification developed with support from the Elastic user community. ECS defines a common set of fields to be used with data in Elasticsearch, such as logs and metrics.

> ECS enables and encourages the normalization of event data, so that you can better analyze, visualize, and correlate the information represented in your events.  This drives efficiency in your operations.

ebeahan commented:
Thanks all for the reviews! I've reworked the doc a bit to incorporate the feedback.

* I split the mapping example into three tables instead of two. However, after splitting the processor-populated fields into its own table, it may be more detail than necessary for a `Getting Started` doc. I'm interested in other's feedback.

    1. The values extracted verbatim from the original log
    2. `ecs.*` and `event.*` metadata fields
    3. Additional fields populated by a processor

* Shifted the search and visualization sections up higher.
* Reworked the intro to incorporate the review feedback and align with the new section ordering.
* Added the links to the field reference guides at the bottom.
* Added link to `Getting Started` to the top-level ECS `index.asciidoc` page.

webmat commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,304 @@
+ECS enables and encourages users to normalize event data in order
+to better analyze, visualize, and correlate their events. Collected events
+can be normalized at ingest time, consistently searched across indices,
+and visualized predictably. When adopting an Elastic solution, such as
```
To make sure this stands out more, I'd start a new paragraph at "When".

Perhaps rephrase "Note that when..."

webmat commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,304 @@
+can be normalized at ingest time, consistently searched across indices,
+and visualized predictably. When adopting an Elastic solution, such as
+https://www.elastic.co/observability[Observability] or https://www.elastic.co/security[Security],
+all of this work is already done! Elastic provides an extensive set of
```
```suggestion
all events will map to ECS out of the box. Elastic provides an extensive set of
```

I think it's worth being a little more explicit that "adopting ECS" is not much work with the solutions.

webmat commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,304 @@
+
+// ==============================================================
+
+| user_agent.os_name
```
This should be `user_agent.host.os.name` ;-)

ebeahan replied:
Good 👀 thank you! Updating to `user_agent.os.name`

webmat commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,304 @@
+
+// ==============================================================
+
+| user_agent.os.version
+| 83.0.4103.106
+| `user_agent`
```
copy/pasta vs row above?

webmat commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,304 @@
+  another field
+* Event collectors, such as https://www.elastic.co/guide/en/beats/libbeat/current/beats-reference.html[Beats], can enrich
+  each event with metadata from the machine's hosting provider (cloud) and/or from the host machine (host).
+
```
Perhaps preface this final table with something like

> Here are some examples of additional fields added by metadata or parser processors.

Also I think we could trim the list of fields we list in the table, to keep it short. A few of each is enough to illustrate, let's keep those that will speak to people most. Perhaps remove

* host.mac
* host.os.build
* either host.os.family or host.os.platform
* user_agent.device.name

ebeahan replied:
Good idea. I was worried the third table was maybe going a little _too_ deep for a getting started section, but I think scaling back is a good way to help address. 👍

webmat commented on docs/using-getting-started.asciidoc:
```diff
@@ -0,0 +1,283 @@
+
+// ==============================================================
+
+| user_agent.os.version
```
That's a Chrome version! 😂 

```suggestion
| user_agent.version
```

![Laser Eyes!](https://webmat.keybase.pub/funny-stuff/dalai_lasers.gif)

ebeahan replied:
😂 It also was missing the final column and wasn't showing up in the rendered doc either! Cleaned up on the latest commit.

webmat commented:
I just noticed we hadn't backported this to 1.5. We should do so, in order for it to appear in the "current" docs now, rather than when we release 1.6.

webmat commented:
When we do backport, we need to adjust the link from Elastic Security to Elastic SIEM.

ebeahan commented:
Thanks for the reminder. I'll work on it 👍
